### DISCUSSION TITLE
How do I perform an action when I receive a specific message?

### DISCUSSION THREAD

**Discussion by ghost** [nicotine-plus/nicotine-plus#2224] [Q&A] [answered]
I [changed](https://lemmy.ml/post/338552) the Leech Detector to ban instead of warn. I would like to unban when I get the message "Added files".

How can I call `self.core.network_filter.unban_user(user)` when I get that message from the user?

**Comment by slook** [comment, marked as answer]
You capture a `incoming_private_chat_notification()` such as:

```
def incoming_private_chat_notification(self, user, line): 
  
    if line.lower() != "added files": 
        return 
  
    self.core.network_filter.unban_user(user)
```
